src/CMakeLists.txt

```diff
@@ -34,6 +34,34 @@ if (HAVE_LIBPFM)
   target_compile_definitions(benchmark PRIVATE -DHAVE_LIBPFM)
 endif()
 
+# pthread affinity, if available
+include(CheckCXXSourceCompiles)
```

Comment:
could we extract this into a separate file in [cmake/](https://github.com/google/benchmark/tree/main/cmake) ?